pkg/controller/service/servicecontroller.go

```diff
@@ -339,14 +339,23 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 
 		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
 
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, api.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-
-		err := s.createLoadBalancer(service, namespacedName)
+		_, exists, err := s.balancer.GetLoadBalancer(s.loadBalancerName(service), s.zone.Region)
```

Comment:
I use openstack as the cloudprovider in my cluster. It happened that `s.createLoadBalancer(service, namespacedName)` fail because of conflicting pool name. Which means, k8s reported that load balancer creating fail and continuously retry with the same response. While, in neutron, the pool has been created successfully already for the first time.
The reason k8s service controller got failure response for the first time is caused by network flapping which cause the following retry.